serialization/writer.py

```diff
@@ -0,0 +1,95 @@
+    'percent', 'ampersand', 'asterisk', 'question', 'at'
+]
+
+FLAGS = flags.FLAGS
+flags.DEFINE_string("protodir", None,
+                    "Directory where all the protos should be saved")
+flags.DEFINE_string("pickledir", None,
+                    "Directory where all the pickles are saved")
+flags.mark_flag_as_required("protodir")
+flags.mark_flag_as_required("pickledir")
+flags.DEFINE_integer("numexamples", 1000,
+                     "Approx number of glyphs in each protobuf file")
+
+
+def pkl2protos(protoDir, pickleName, fileNum):
+    '''
+    This function takes one pickle and proto directory and puts all of its glyphs
+    into protos in the proto dir
+
+    Parameters
+    ----------
+    proto_dir: the directory where all the protos should be saved
+
+    pickle: the location of the pickle to be put into protos
+    '''
+    try:
+        with open(pickleName, 'rb') as fontString:
+            fontDict = pickle.load(fontString)
+            for glyph in fontDict:
+                # basically we are going to flatten everything into just an array of
+                # points
+                # we will keep track of the location of where each contour stops so we
+                # can reconstruct on the other end.
+                proto = font_pb2.glyph()
+                if glyph in CHARS:
+                    contours = fontDict[glyph]
+                    num_contours = len(contours)
+                    points = []
+                    contour_locations = []
+                    for _, c in contours.items():
+                        contour_locations.append(len(c))
+                        for curve in c:
+                            for point in curve:
+                                points += point
+                    #write it in
+                    newGlyph = proto.glyph.add()
+                    newGlyph.num_contours = num_contours
+                    points = list(points)
+                    newGlyph.bezier_points.extend(points)
+                    newGlyph.contour_locations.extend(contour_locations)
+                    newGlyph.font_name = pickleName
+                    newGlyph.glyph_name = glyph
+                    #save it up
+
+                    with open("{}/{}{}".format(protoDir, glyph, fileNum),
+                              'ab') as f:
+                        f.write(proto.SerializeToString())
+
+    except Exception as E:
+        print(pickleName, E, sys.exc_info()[0])
+
+
+def main(argv):
+    numexamps = int(FLAGS.numexamples)
+    protoDir = FLAGS.protodir
+    pickleDir = FLAGS.pickledir
+
+    #find all the pickles
+    pickles = glob.glob('{}*.p'.format(pickleDir))
+    # iterate through them and keep track of the example we are up to
+    count = 0
+    fileNum = 1
+    for pickle in pickles:
+        if count > numexamps:
+            count = 0
+            fileNum += 1
+        pkl2protos(protoDir, pickle, fileNum)
```

Comment:
Do we want to keep only one font glyph per protobuf? We could alternately push more in but i guess that would require rethinking this a little bit. You're not counting how many of each glypth we have, just how many fonts we've unpickled.

Reply:
It would be a little more difficult to get the exact number of glyphs we have in each file, we would need to keep a separate count for each glyph.